### DISCUSSION TITLE
Decentralisons SMTP pour le bien commun · poolp.org

### DISCUSSION THREAD

**Discussion by poolpOrg** [poolpOrg/poolp.org#112] [Articles]

**Comment by B4rb3rouss** [comment]
Merci pour cet article qui pointe vers la configuration en anglais.
Pour information, je me suis permis d'en reprendre certaines lignes dans [un guide](https://ybad.name/ah/#heberger-son-courrier-electronique) plus complet, en ajoutant opensmtpd-extras pour partager des tables "passwd" avec dovecot (je ne connais pas de meilleurs moyens).
J'en profite aussi pour te remercier de tout ton travail sur OpenBSD :)

**Comment by poolpOrg** [comment]
Top !

Pour les tables "passwd" en fait ça dépends de quelles fonctionnalités tu veux utiliser dans Dovecot.

Si tout ce que tu veux c'est authentifier des utilisateurs virtuels et que tu ne te sers que du username et du password, alors tu peux créer un simple fichier qui contient deux colonnes séparées par un `:` genre:

    $ cat /etc/mail/accounts.txt
    gilles:$2b$09$1l68v6AhM9cd8v6iWtBEnuNio1m.bSwnOYA750zkL7Kq9YXPEyIuK

Côté OpenSMTPD, tu peux utiliser une table tout ce qu'il y a de plus normale avec ce fichier en backend:

    $ grep accounts /etc/mail/smtpd.conf
    table accounts "/etc/mail/accounts.txt"
    listen on all tls auth <accounts>

Côté Dovecot, tu peux utiliser le fichier en backend pour passdb et te servir du format `%u` pour userdb:

    $ cat auth-custom.conf.ext                                                                                                                                                                                                                                                                                     
    passdb {
      driver = passwd-file
      args = scheme=CRYPT /etc/mail/accounts.txt
    }
    
    userdb {
      driver = static
      args = uid=_vusers gid=_vusers home=/var/maildir/%u
    }

**Comment by B4rb3rouss** [comment]
* Gilles Chehade <notifications@github.com> le [15-12-2019 23:11:34 -0800]:
> Top !
> 
> Pour les tables "passwd" en fait ça dépends de quelles fonctionnalités tu veux utiliser dans Dovecot.

Dans le cadre de mon tuto, pas besoin de quotas ou autres
fonctionnalités supplémentaires. C'est pour de l'auto-hébergement ou un
nombre restreint d'utilisateurs (moins de 1000 on va dire). 
C'est tout bête finalement quand c'est toi qui le décrit. J'ai eu beau
éplucher la doc de dovecot et la man table(5)...

J'utilise des utilisateurs virtuels mais héberge plusieurs noms de
domaine à titre perso. Ta solution fonctionne en précisnat 

    batman@domain.tld:$2B$.....

Merci ! :)

**Comment by hucste** [comment]
Gilles, si tu permets, et je serais étonné du contraire, je vais te faire une proposition de relecture - promis en tenant compte de ta spécificité de nouvelle ligne, après virgule :p

**Comment by poolpOrg** [comment]
@hucste  avec plaisir :-)

**Comment by marcespie** [comment]
"la plupart de certaines de ces entreprises" ← faudrait choisir !!!!
